libnd4j/include/ops/declarable/platform/cudnn/ctcloss.cu

```diff
@@ -85,17 +69,16 @@ namespace platforms {
         cudnnStatus_t err = CUDNN_STATUS_SUCCESS;
         callCudnnIfNoErr(err, cudnnSetStream, *handle, *context.getCudaStream());
 
-        cudnnCTCLossDescriptor_t  ctcLossDesc;
-        cudnnTensorDescriptor_t probsDesc = nullptr;
-        cudnnTensorDescriptor_t gradsDesc = nullptr;
-        callCudnnIfNoErr(err, cudnnCreateCTCLossDescriptor, &ctcLossDesc);
-        callCudnnIfNoErr(err, cudnnSetCTCLossDescriptorEx, ctcLossDesc, CUDNN_DATA_FLOAT, CUDNN_LOSS_NORMALIZATION_SOFTMAX, CUDNN_PROPAGATE_NAN);
-        callCudnnIfNoErr(err, cudnnCreateTensorDescriptor, &probsDesc);
-        callCudnnIfNoErr(err, cudnnSetTensorNdDescriptor, probsDesc, cudnnDataType(probs.dataType()), probs.rankOf() , dims, strides);
-        if(!grads.isEmpty()){
+        CTCLossDesc  ctcLossDesc;
+        CudnnTensor probsDesc, gradsDesc(false);
+        bool calcGrads = !grads.isEmpty();
+        ctcLossDesc.set( err, CUDNN_DATA_FLOAT, CUDNN_LOSS_NORMALIZATION_SOFTMAX, CUDNN_PROPAGATE_NAN);
```

Comment:
Any reason for the hard coded data type here? Shouldn't we have a conversion for the known data type? Either way, a comment here would be good as well.

Reply:
my bad, I do not know why I did not put it as a derived datatype for the future. that is a lesser confusion.

we have a CTC loss (maybe in the future, cudnn-they consider other types as well)  is restricted (only for Float32 and input32). I'm thinking of refactoring ctc Loss again ( it needs if def sections and proper comments so it was not confusing ). and also adding v8 api calls .

Reply:
now it's determined by input type. but still, "Cudnn CTC loss" is restricted (only float32). I'll think of relaxing it and adding the conversion.

Reply:
I'll address all these next time when I'll refactor CUDNN again.